src/poprox_recommender/default.py

```diff
@@ -245,6 +309,44 @@ def compute_similarity_matrix(todays_article_vectors):
                 similarity_matrix[i, j] = similarity_matrix[j, i] = np.dot(value1, value2)
     return similarity_matrix
 
+def find_topic(past_articles: List[Article], article_id: UUID):
+    # each article might correspond to multiple topic
+    for article in past_articles:
+        if article.article_id == article_id:
+            return [mention.entity.name for mention in article.mentions]
+
+def normalized_topic_count(topic_count_dict):
+
+    normalized_count = {}
+    total_count = sum(topic_count_dict.values())
+    for topic in topic_count_dict:
+        normalized_count[topic] = topic_count_dict[topic] / total_count
+    return normalized_count
+
+def user_topic_preference(
+    past_articles: List[Article],
+    click_histories: List[ClickHistory]
+):
+    '''Topic preference only based on click history'''
+    user_preference_dict = {}
+    for click_history in click_histories:
+        account_id = click_history.account_id
+        clicked_articles = click_history.article_ids # List[UUID]
+
+        topic_count_dict = {}
```

Comment:
If you make this a `defaultdict` you can save yourself the step of checking whether keys are present or not, because all keys will get the supplied default value (zero, in this case) when they're accessed for the first time:
```
from collections import defaultdict

topic_count_dict = defaultdict(lambda: 0)
...
for topic in clicked_topics:
    topic_count_dict[topic] += 1

Reply:
Good point, updated!